src/activities/find-all-docmaps.ts

```diff
@@ -1,16 +1,34 @@
 import axios from 'axios';
 import { DocMap } from '@elifesciences/docmap-ts';
+import { MD5 } from 'object-hash';
 
 type DocMapIndex = {
   docmaps: DocMap[],
 };
 
-export const findAllDocmaps = async (docMapIndex: string): Promise<DocMap[] | undefined> => {
-  const { data, status } = await axios.get<DocMapIndex>(docMapIndex);
+type FindAllDocMapsResult = {
+  docMaps: DocMap[],
+  hashes: string[],
+};
+
+// TO-DO: Replace this with find changed docmaps
+export const findAllDocmaps = async (hashes: string[], docMapIndex: string): Promise<FindAllDocMapsResult | undefined> => {
+  const docMapRes = await axios.get<DocMapIndex>(docMapIndex);
 
-  if (status === 200) {
-    return data.docmaps;
+  if (typeof docMapRes !== 'object' || docMapRes.status !== 200) {
```

Comment:
While this is improveing current error handling as it was, this is worth a conversation I think, and it's more than just this bit - how do we handle errors in import?

Personally, I'd be of the opinion that we don't explicitly return "successfully" unsuccessful here - non 200 errors can be for a number of reasons. Then we can allow the upper workflow to define a [retry policy](https://docs.temporal.io/application-development/features?lang=typescript#activity-retries). This way, transient issues are retried and recovered from rather than explicitly stopped. Then when the retry failure occurs, we can notify, stop the workflow, etc at the workflow level. Thoughts on that as an approach?

For now, it's fine to extend the failure detection, but might be worth a follow on

Reply:
This is definitely a follow-up, so much to the point I was thinking of taking this [task](https://github.com/elifesciences/enhanced-preprints-issues/issues/602) on afterwards that addresses it 😅